### ISSUE TITLE
Elastic flag leads to core dump

### ISSUE THREAD

**Issue by P-Diddy-P** [algbio/founderblockgraphs#10]
Hello, I wanted to check some stuff regarding founder block graphs and tried reproducing the results of [this paper](https://arxiv.org/pdf/2005.09342.pdf). The reproduction is using about the same data - all covid19 sequences up to 24.4.2020, MSA using ViralMSA with a similar version (attached below). The number of sequences isn't identical but relatively close (1431 as opposed to 1484 in the paper).

Running the tool without the `elastic` flag, the results are as expected. However, with `elastic` flag on, the program is aborted following an error line seen [here](https://github.com/algbio/founderblockgraphs/blob/master/founderblockgraph.cpp#L1514).

My questions are:
1. I've used only `elastic` flag running the reproduction, are there any other options I might've missed which would make this work?
2. Is the `elastic` flag viable for use in the current state of the project?
3. Is this project still being actively worked on to implement the later findings by your group regarding EFGs and semi-repeat free segmentations?

Thanks in advance!

[covid_sequences.fasta.zip](https://github.com/algbio/founderblockgraphs/files/14321542/covid_sequences.fasta.zip)

**Comment by nrizzo** [comment]
Hey, thanks for spotting this!

Short answer: I have been developing semi-repeat-free EFG construction using both options `--elastic` and `--gfa`. I will document this construction ASAP and soonish we will publish pattern matching on semi-repeat-free EFGs (and other useful tools). So, for the time being, we do not have pattern matching, and I will disable the old output for the `--elastic` option (i.e. you just get the GFA graph output).

For your dataset, I would run the tool as follows:
```
founderblockgraph --elastic --input=covid_sequences.fasta.aln --gfa --output-paths --output=efg.gfa --threads 8 --ignore-chars="N"
```
On my machine, this generates a nice graph with the input sequences as paths that you can load in [BandageNG](), for example. See the long answer below to validate the graph.

Thanks again for checking out the repo, and please get in touch if you have more questions!
--Nicola

Longer answer: For the EFG construction, I have implemented the min max length segmentation from [our last paper](https://doi.org/10.1016/j.tcs.2023.114269 "Elastic founder graphs improved and enhanced") (Algs 1 and 4), with a few tricks that help the segmentation:
- each sequence starts and ends (only conceptually) with a unique character
- long starting/ending gap runs are not a problem
- if we promise not to query some character (e.g. `N`s), we can consider each of them as unique

This is helpful for your dataset, since you have gaps and `N`s: for example, MT293547.1 contains only 63 bases! The indexing, that we did not implement yet, should work one the concatenated edges in a similar way as before.

To answer 2. and 3., you can (sort of) check that the output graph contains the input sequences with pangenomic tools like [`vg`](https://github.com/vgteam/vg):
```
# convert in vg format and output paths
# (characters different from ACGT become N!)
vg convert --gfa-in efg.gfa --packed-out > efg.vg
vg paths --vg efg.vg --extract-fasta > efgpaths.fasta
# check if sequences are the same, fixing bp differences and format
diff <(grep -v '^>' input/covid_sequences.fasta.aln | tr -d "-" | tr -C "[ACGT\n]" "N" | sort) <(seqtk seq efgpaths.fasta | grep -v '^>' | sort)
```
and soonish we will also publish validation of the indexing property!

**Comment by nrizzo** [comment]
Patched in commit 174d3df.
